### DISCUSSION TITLE
Exception in Razor page

### DISCUSSION THREAD

**Discussion by Herdubreid** [alpinejs/alpine#2151] [1. Help]
Alpine throws an exception on the `x-show` attribute when the content has `@` content.

Example (from The [BlazorAlpine](https://github.com/Herdubreid/BlazorAlpine) github project:
``` razor
<h4>From Example</h4>
<div x-data="{ open: false }">
    <button @click="open = ! open">Toggle Content</button>
 
    <div x-show="open">
        Content...
    </div>
</div>

<!-- This throws an exception -->
<h4>With Blazor Tag</h4>
<div x-data="{ open: false }">
    <button @click="open = ! open">Toggle Content</button>
 
    <div x-show="open">
        @Content
    </div>
</div>

@code {
    string Content { get; set; } = "Content...";
}
```
I think Alpine is a great complementary framework for Blazor, even with this limitation!

It would be fantastic if there is a solution/workaround for this.

**Comment by danddanddand** [comment]
I'm not familiar with Blazor but more information about error you are experiencing and link to example would be useful.

**Comment by SimoTod** [comment]
I don't think it's correct: it's Blazor causing the error not Alpine. 
As per Blazor documentation, you must escape your `@`s: 
https://docs.microsoft.com/en-us/aspnet/core/mvc/views/razor?view=aspnetcore-3.0#razor-syntax

i.e. `@@click`

**Reply by Herdubreid** [reply]
It's one of those potato/potatoe issues - and I didn't log this as a bug :).
alpine throws the exception, but I agree that Blazor is the culprit.

It's easy to replicate this (https://github.com/Herdubreid/BlazorAlpine as above)

I can see Alpine relieve so many Blazor headache's, but I don't have a clue how/if this can be solved.

I'm happy to work with you on this, however I'm an application programmer so I just want things to work :)

**Reply by danddanddand** [reply]
If what SimoTod said is correct looks like you can just escape the @ or use x-on:click instead?

**Reply by Herdubreid** [reply]
The issue is not about the @ attribute, but you're right that you need to use x- instead.
alpine throws an exception on x-show with @ content.
![image](https://user-images.githubusercontent.com/21098321/135484942-d8ed7f6c-d018-4fd8-a07c-999d2ea90747.png)

**Reply by SimoTod** [reply]
I don't have a .net setup so i can't try but have you tried using `@@click` as required by Blazor? I don't think the issue is `@content`, alpine shouldn't even see it. Especially if using x-on:click works.

**Reply by Herdubreid** [reply]
I've tested this a bit more (`@@click` doesn't work at all), and found that this behaviour appears to be introduced in version 3.0.0.  Version 2.8.2 doesn't have this issue, but version 3.0.0 errors as follows:
`Uncaught (in promise) TypeError: i is not a function
    at alpine.js:1677`
![image](https://user-images.githubusercontent.com/21098321/135570569-f7421437-b802-4b09-87b5-bfedb8ea6905.png)

**Comment by fbragason** [comment]
I was just re-visiting this and I can't replicate the exception!

There have been Visual Studio upgrades and .NET6 has been published in the last couple of months, so there have been changes on the razor side.

I'm not going to attempt to dig into what fixed this, just happy I can now use Alpine with Blazor!
